### ISSUE TITLE
[question] friend function defined within class?

### ISSUE THREAD

**Issue by harsszegi** [xoofx/CppAst.NET#86]
Hi,
does CppAst support friend functions defined within the class itself? E.g. something like this:

class Foo
{
public:
   friend bool operator==(const Foo & lhs, const Foo & rhs);
};

I fail to see the operator== function anywhere, neither within the class's functions (that's kinda obvious), nor between the functions in the namespace.
What do I miss here?
Thanks,

**Comment by xoofx** [comment]
> I fail to see the operator== function anywhere, neither within the class's functions (that's kinda obvious), nor between the functions in the namespace.

Dunno. CppAst.NET was initially not developed to support C++, so it might miss many stuffs like this. You would have to debug/add support for it in a PR.

**Comment by fangfang1984** [comment]
Yes, friend functions is not contain in CppClass now, i try to add it in our local branch, it will be PR to master recently.

**Comment by xoofx** [comment]
Fixed via commit 320bc7e
